plugins/woocommerce/src/Internal/Admin/WcPaySubscriptionsPage.php

```diff
@@ -74,6 +71,62 @@ public function register_subscriptions_page() {
 		}
 	}
 
+	/**
+	 * Returns true if the store is eligible for the WooCommerce subsciptions empty state experiment.
+	 *
+	 * @return bool
+	 */
+	private function is_store_experiment_eligible() {
+		// Ineligible if WooCommerce Payments OR an existing subscriptions plugin is active.
+		$active_plugins         = PluginsHelper::get_active_plugin_slugs();
+		$plugin_ineligible_list = array(
+			'woocommerce-payments',
+			'woocommerce-subscriptions',
+			'subscriptions-for-woocommerce',
+			'subscriptions-for-woocommerce-pro',
+			'yith-woocommerce-subscription',
+		);
+		foreach ( $plugin_ineligible_list as $plugin_slug ) {
+			if ( in_array( $plugin_slug, $active_plugins, true ) ) {
+				return false;
+			}
+		}
+
+		// Ineligible if store address is not compatible with WCPay Subscriptions (US).
+		$store_base_location = wc_get_base_location();
+		if ( empty( $store_base_location['country'] ) || 'US' !== $store_base_location['country'] ) {
+			return false;
+		}
+
+		// Ineligible if store has not been active for at least 6 months.
+		if ( ! WCAdminHelper::is_wc_admin_active_in_date_range( 'month-6+' ) ) {
+			return false;
+		}
+
+		// Ineligible if store has not had any sales in the last 30 days.
+		$store_recent_sales_count = $this->get_store_recent_sales_count();
+		if ( $store_recent_sales_count < 1 ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns the number of store recent sales as per the experiment eligibility criteria.
+	 */
+	private function get_store_recent_sales_count() {
+		// Get orders that were paid in the last 30 days.
+		$orders = wc_get_orders(
+			array(
+				'date_paid' => '>' . strtotime( '-30 days' ),
+				'status'    => array( 'wc-completed' ),
+			)
+		);
```

Comment:
Because for this purpose we only need to check for the existence of at least 1 order for the store to be eligible, we could limit this query to `1`. eg `'limit' => 1`

It's just a little more performant. By default the limit arg gets set to the site's "posts_per_page" var which is 10 normally so this isn't a big deal, but that value can be changed to anything and we could be getting 100 orders only to check if there is 1 for example. 

Another tiny improvement is to only get the order IDs rather than the order objects (via `'return' => 'ids'`). By default `wc_get_orders()` returns the full order objects. When WC loads each order it has to pull data from other tables and potentially third-party plugins can get involved via the action hooks that fire, so in cases like this where we're running this query to make a determination about the store, we can tighten the query just a little to limit the scope.

Reply:
Just noting that these changes change the original design of this function so it might need to also change into bool return rather than a count but I'll leave that up to you to decide 😃 .

Reply:
Thanks @james-allan!

I'll implement this change to tighten the query to the bare essentials.

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/32867/commits/0eb8d95bfb0d99acfd62a760c6d508f8f4278a16